neurodamus/core/coreneuron_configuration.py

```diff
@@ -0,0 +1,129 @@
+import os
+import logging
+from pathlib import Path
+from ._utils import run_only_rank0
+from . import NeurodamusCore as Nd
+
+
+class _CoreNEURONConfig(object):
+    """
+    The CoreConfig class is responsible for managing the configuration of the CoreNEURON simulation.
+    It writes the simulation / report configurations and calls the CoreNEURON solver.
+    """
+    sim_config_file = "sim.conf"
+    report_config_file = "report.conf"
+    output_root = "output"
+    datadir = f"{output_root}/coreneuron_input"
+    default_cell_permute = 0
+    artificial_cell_object = None
+
+    # Instantiates the artificial cell object for CoreNEURON
+    # This needs to happen only when CoreNEURON simulation is enabled
+    def instantiate_artificial_cell(self):
+        self.artificial_cell_object = Nd.CoreNEURONArtificialCell()
+
+    @run_only_rank0
+    def write_report_config(
+            self, report_name, target_name, report_type, report_variable,
+            unit, report_format, target_type, dt, start_time, end_time, gids,
+            buffer_size=8):
+        import struct
+        num_gids = len(gids)
+        logging.info(f"Adding report {report_name} for CoreNEURON with {num_gids} gids")
+        report_conf = Path(self.output_root) / self.report_config_file
+        report_conf.parent.mkdir(parents=True, exist_ok=True)
+        with report_conf.open("ab") as fp:
+            # Write the formatted string to the file
+            fp.write((" ".join(map(str, [
+                report_name,
+                target_name,
+                report_type,
+                report_variable,
+                unit,
+                report_format,
+                int(target_type),
+                float(dt),
+                float(start_time),
+                float(end_time),
+                num_gids,
+                buffer_size
+            ])) + "\n").encode())
```

Comment:
Why dont we use the same formatting as in the source? I'd find easier to maintain and likely more compatible
```suggestion
            fp.write("%s %s %s %s %s %s %d %lf %lf %lf %d %d\n" % (
                report_name,
                target_name,
                report_type,
                report_variable,
                unit,
                report_format,
                target_type,
                dt,
                start_time,
                floatend_time,
                num_gids,
                buffer_size,
            )
```

Reply:
Sorry, the format string should be a byte string, i.e. `b"%s %s. ..."`